tsconfig.tests.json

```diff
@@ -0,0 +1,12 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "strict": true,
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "noImplicitReturns": true,
+    "noFallthroughCasesInSwitch": true
```

Comment:
Why these specific options, are they just rules you think should be enforced or is it a recommended pattern?

Reply:
We could definitely add more to be more strict, but these were some that I have found to be useful in some of my own TypeScript projects. If I recall correctly they were adapted from a default strict tsconfig. Here's what they do:

```
    /* Strict Type-Checking Options */
    "strict": true /* Enable all strict type-checking options: noImplicitAny, strictNullChecks, strictFunctionTypes, strictPropertyInitialization, noImplicitThis, alwaysStrict */,
    "noUnusedLocals": true /* Report errors on unused locals. */,
    "noUnusedParameters": true /* Report errors on unused parameters. Requires TypeScript version 2.0 or later.*/,
    "noImplicitReturns": true /* Report error when not all code paths in function return a value. */,
    "noFallthroughCasesInSwitch": true /* Report errors for fallthrough cases in switch statement. */,
```

Reply:
Great, thanks for providing detail for each rule